src/wasm/wasm-binary.cpp

```diff
@@ -1090,6 +1090,7 @@ Type WasmBinaryBuilder::getType() {
   // Single value types are negative; signature indices are non-negative
   if (type >= 0) {
     // TODO: Handle block input types properly
+    assert(size_t(type) < signatures.size());
```

Comment:
I think this would be invalid input, so we should error on it with `throwError`?

Also, how about using `signatures::size_type`?

Reply:
Actually, the syntax necessary to use `signatures::size_type` is so verbose that I would rather just use `size_t`. I would have to write `if (std::vector<Signature>::size_type(type) < ...`.